doc/md/tutorial-grpc-service-generation-options.md

```diff
@@ -0,0 +1,51 @@
+---
+id: grpc-service-generation-options
+title: Configuring Service Method Generation
+sidebar_label: Service Generation Options
+---
+By default, entproto will generate CRUD service methods for an `ent.Schema` annotated with `ent.Service()`. Method generation can be customized by including the argument `entproto.Methods()` in the `entproto.Service()` annotation. `entproto.Methods()` accepts bit flags to determine what service methods should be generated. The flags include:
+```go
+// Generates a Create gRPC service method for the entproto.Service.
+entproto.MethodCreate
+
+// Generates a Get gRPC service method for the entproto.Service.
+entproto.MethodGet
+
+// Generates an Update gRPC service method for the entproto.Service.
+entproto.MethodUpdate
+
+// Generates a Delete gRPC service method for the entproto.Service.
+entproto.MethodDelete
+
+// Generates all service methods for the entproto.Service.
+// This is the same behavior as not including entproto.Methods.
+entproto.MethodAll
+```
+To generate a service with multiple methods, bitwise OR the flags.
+
+
+To see this in action, we can modify our ent schema. Let's say we wanted to prevent our gRPC client from mutating entries. We can accomplish this by modifying `ent/schema/user.go`:
+```go
+func (User) Annotations() []schema.Annotation {
+	return []schema.Annotation{
+		entproto.Message(),
+		entproto.Service(
+			entproto.Methods(                   // <-- add this
+				entproto.MethodCreate |         // <-- ...
+                entproto.MethodGet              // <-- ...
```

Comment:
please use https://docusaurus.io/docs/markdown-features/code-blocks#multiple-line-highlighting for highlighting important lines instead :-)